controllers/storageconsumer/storageconsumer_upgrade_controller.go

```diff
@@ -121,8 +135,30 @@ func (r *StorageConsumerUpgradeReconciler) Reconcile(ctx context.Context, reques
 		return reconcile.Result{}, err
 	}
 
-	if err = r.reconcileStorageConsumer(ctx, storageCluster, storageConsumer, consumerConfigMapName); err != nil {
-		return reconcile.Result{}, err
+	if internalConsumer {
+		newConsumer := &ocsv1alpha1.StorageConsumer{}
+		newConsumer.Name = defaults.LocalStorageConsumerName
+		newConsumer.Namespace = storageConsumer.Namespace
+		newConsumer.Spec.Enable = true
```

Comment:
this'll create issues afaics, enable will trigger consumer controller since no CM is created the defaults values will be propagated and consumer will be made as controlling owner.

Reply:
The consumer configmap is created above in the upgrade controller